packages/lib/src/interactions/hooks.ts

```diff
@@ -23,22 +22,22 @@ export function useMoveCameraTo() {
     (x: number, y: number) => {
       const { position } = camera;
 
-      const { topRight } = getCameraFOV(camera);
-      const cameraLocalBounds = topRight.sub(position);
+      const { topRight, bottomLeft } = getCameraFOV(camera);
+      const width = Math.abs(topRight.x - bottomLeft.x);
+      const height = Math.abs(topRight.y - bottomLeft.y);
 
-      const xBound = Math.max(visWidth / 2 - cameraLocalBounds.x, 0);
-      const yBound = Math.max(visHeight / 2 - cameraLocalBounds.y, 0);
-
-      position.set(
-        clamp(x, -xBound, xBound),
-        clamp(y, -yBound, yBound),
-        position.z
+      const clampedPosition = clampRectangleToArea(
+        new Vector2(x, y),
```

Comment:
Awesome! I think this even opens the door, in a future PR, to let `moveCameraTo` and `clampRectangleToArea` accept `Vector2 | Vector3`.

Only minor thing that bugs me is the naming of `clampRectangleToArea`, which doesn't fit the return type (`Vector2`). Perhaps `clampPositionToArea` would make more sense?